### PR TITLE
Add externalContracts explanations on hooks doc

### LINKED ISSUE #35: Add details / section about adding externalContracts
Once we merge https://github.com/scaffold-eth/scaffold-eth-2/pull/592, it would be really helpful if we add section / page on how to add / interact with external contract  maybe we add it #25 section

### PR DESCRIPTION
**Pull request by Pabl0cks**
I've just added a initial explanation on how to add external contracts to `externalContracts.ts` file, in order to use Scaffold ETH-2 hooks on them. May need a first tech review from @technophile-04 the explanations are correct.

Was not sure about adding a full mock test example at the end of the doc, since it helped to me I've finally added it, but we can ✂ if you see not necessary.

Closes #35

### CODE REVIEW

File: docs/hooks/hooks.md [hunk 1]
```diff
@@ -6,4 +6,1335 @@ sidebar_position: 4
 
 Scaffold-ETH 2 provides a collection of custom React hooks designed to simplify interactions with your deployed smart contracts. These hooks are wrappers around `wagmi`, automatically loading the necessary contract ABI and address. These hooks offer a simplified and streamlined interface for interacting with your smart contracts.
 
-If you need to interact with external contracts, you can use `wagmi` directly, or add external contract data to your `deployedContracts.ts` file.
+If you need to interact with external contracts you can use `wagmi` directly, or you can add external contract data to your `externalContracts.ts` file, which would let you use Scaffold-ETH 2 hooks with autocompletion to interact with them.
```
Comment (carletex) [inline docs/hooks/hooks.md:9]: My suggestion:

"If you need to interact with external contracts (i.e. not deployed with your SE-2 instance) you can add external contract data to your `packages/nextjs/contracts/externalContracts.ts` file, which would let you use Scaffold-ETH 2 hooks"

File: docs/hooks/hooks.md [hunk 2]
```diff
@@ -6,4 +6,1335 @@ sidebar_position: 4
 
 Scaffold-ETH 2 provides a collection of custom React hooks designed to simplify interactions with your deployed smart contracts. These hooks are wrappers around `wagmi`, automatically loading the necessary contract ABI and address. These hooks offer a simplified and streamlined interface for interacting with your smart contracts.
```
Comment (carletex) [inline docs/hooks/hooks.md:7]: I think we can change the text here too:

"These hooks are wrappers around Wagmi, an easy-to-use interface with typescript autocompletions for reading from, writing to, and monitoring events emitted by your smart contracts."

(from the SE2 landing)

File: docs/hooks/hooks.md [hunk 3]
```diff
@@ -6,4 +6,1335 @@ sidebar_position: 4
...
+      abi: [...],
+    },
+    USDT: {
+      address: "0x...",
+      abi: [...],
+    },
+} as const;
+```
+
+<details>
```
Comment (carletex) [inline docs/hooks/hooks.md:38]: Even if this is folded, it seems like too much, and I think the example above is enough.

**Comment by carletex** [comment]
Great stuff @Pabl0cks , thanks!

Left some comments with my views

**Comment by Pabl0cks** [comment]
Thanks for the review @carletex, love all the suggestions! Just pushed them 🙌♥ 

Indeed the mock example was a bit too much, even took a while to unfold on my computer haha 

Maybe if we see people needing help with their `externalContracts.ts` file we can just share them the mock example with a [gist](https://gist.github.com/Pabl0cks/4235cc0aec6f08e690b7d658c79d77dd). But probably people using external contracts won't have problems with `abi` format.

**Pull request merged**